tsdb/engine/tsm1/compact_test.go

```diff
@@ -4201,58 +4201,178 @@ func TestEnginePlanCompactions(t *testing.T) {
 		},
 		[]tsm1.FileStat{
 			{
-				Path: "01-05.tsm",
-				Size: 256 * 1024 * 1024,
+				Path: "01-05.tsm1",
+				Size: 2048 * 1024 * 1024,
 			},
 			{
-				Path: "02-05.tsm",
-				Size: 256 * 1024 * 1024,
+				Path: "01-06.tsm1",
+				Size: 2048 * 1024 * 1024,
 			},
 			{
-				Path: "03-05.tsm",
-				Size: 256 * 1024 * 1024,
+				Path: "01-07.tsm1",
+				Size: 2048 * 1024 * 1024,
 			},
 			{
-				Path: "04-05.tsm",
-				Size: 256 * 1024 * 1024,
+				Path: "01-08.tsm1",
+				Size: 1048 * 1024 * 1024,
+			},
+			{
+				Path: "02-05.tsm1",
+				Size: 2048 * 1024 * 1024,
+			},
+			{
+				Path: "02-06.tsm1",
+				Size: 2048 * 1024 * 1024,
+			},
+			{
+				Path: "02-07.tsm1",
+				Size: 2048 * 1024 * 1024,
+			},
+			{
+				Path: "02-08.tsm1",
+				Size: 1048 * 1024 * 1024,
+			},
+			{
+				Path: "03-04.tsm1",
+				Size: 2048 * 1024 * 1024,
+			},
+			{
+				Path: "03-05.tsm1",
+				Size: 512 * 1024 * 1024,
 			},
 		},
 	}
-	testBlockCountsAndResults := []struct {
+
+	type testBlockCountsAndResults struct {
 		blockCounts  []int
 		level1Groups []tsm1.PlannedCompactionGroup
 		level2Groups []tsm1.PlannedCompactionGroup
 		level3Groups []tsm1.PlannedCompactionGroup
 		level4Groups []tsm1.PlannedCompactionGroup
 		level5Groups []tsm1.PlannedCompactionGroup
-	}{
+	}
+
+	type testEnginePlanCompactionsRunner struct {
+		name  string
+		files []tsm1.FileStat
+		// Each result is for the different plantypes
+		testBlockCountsAndResults []testBlockCountsAndResults
```

Comment:
WIP I'm thinking about changing this to have either a slice of compaction groups only :thinking: